File: c++/ARRAY/linear_search.cpp
#include<iostream>
using namespace std;
bool search(int arr[], int size, int key)
{
    for(int i=0; i<size; i++)
    {
        if(arr[i]==key)
        {
            return 1;
        }
    }
    return 0;
}

int main()
{
    int arr[100], size, key;
    cout<<"enter size of array: ";
    cin>>size;
    cout<<"enter elements in array: "<<endl;
    for(int i=0; i<size; i++)
    {
        cin>>arr[i];
    }
    cout<<"enter element for search: ";
    cin>>key;
    bool found = search(arr, size, key);
    if(found)
    {
        cout<<"Key is present";
    }
    else
    {
        cout<<"Key is absent";
    }
    return 0;
}

File: javaPractice/SEARCH/LinearSearch.java
public class LinearSearch {
    public static void main(String[] args) {
        int[] arr = { 89, 123, 13, 80, 179, 269, 130 };
        int res = search(arr, 17);
        if(res!=-1)
            System.out.println("Element found at : "+ res);
        else
            System.out.println("Element not found");
    }

    static int search(int[] arr, int key){
        for(int i=0; i<arr.length; i++){
            if(key == arr[i])
                return i;
        }
        return -1;
    }
}
